include/ddc/kernels/splines/splines_linear_problem_2x2_blocks.hpp

```diff
@@ -109,6 +144,59 @@ class SplinesLinearProblem2x2Blocks : public SplinesLinearProblem<ExecSpace>
         }
     }
 
+    /**
+     * @brief Fill a COO version of a Dense matrix (remove zeros).
+     *
+     * [SHOULD BE PRIVATE (GPU programming limitation)]
+     *
+     * Runs on a single thread to garantee ordering.
+     *
+     * @param dense_matrix The dense storage matrix whose non-zeros are extracted to fill the COO matrix.
+     *
+     * @return The COO storage matrix filled with the non-zeros from dense_matrix.
+     */
+    Coo dense2coo(Kokkos::View<double**, Kokkos::LayoutRight, typename ExecSpace::memory_space>
+                          dense_matrix)
+    {
+        Kokkos::View<int*, Kokkos::LayoutRight, typename ExecSpace::memory_space> rows_idx(
+                "ddc_splines_coo_rows_idx",
+                dense_matrix.extent(0) * dense_matrix.extent(1));
+        Kokkos::View<int*, Kokkos::LayoutRight, typename ExecSpace::memory_space> cols_idx(
+                "ddc_splines_coo_cols_idx",
+                dense_matrix.extent(0) * dense_matrix.extent(1));
+        Kokkos::View<double*, Kokkos::LayoutRight, typename ExecSpace::memory_space>
+                values("ddc_splines_coo_values", dense_matrix.extent(0) * dense_matrix.extent(1));
+
+        Kokkos::DualView<std::size_t, Kokkos::LayoutRight, typename ExecSpace::memory_space>
+                n_nonzeros("ddc_splines_n_nonzeros");
+        n_nonzeros.h_view() = 0;
+        n_nonzeros.modify_host();
+        n_nonzeros.sync_device();
+
+        Kokkos::parallel_for(
+                "dense2coo",
+                Kokkos::RangePolicy(ExecSpace(), 0, 1),
+                KOKKOS_LAMBDA(const int) {
+                    for (int i = 0; i < dense_matrix.extent(0); i++) {
+                        for (int j = 0; j < dense_matrix.extent(1); j++) {
+                            double aij = dense_matrix(i, j);
+                            if (Kokkos::abs(aij) >= 1e-14) {
```

Comment:
```suggestion
                            if (Kokkos::abs(aij) >= 1e-14) {
```

I think we can strictly compare to 0 ? The zeros come non-modifications of the matrix, not from the result of a computation ?

Reply:
There is a computation for the top-right corner block during factorization, which is Q^-1*gamma, which look like this:
![image](https://github.com/CExA-project/ddc/assets/22033714/efe55b66-c8f5-4781-94c9-3048726a0d2e)

Reply:
I see, do we loose a lot of performance if we only remove exact zeroes ?

Reply:
Yes (I guess a lot of almost-zero are retained):

![image](https://github.com/CExA-project/ddc/assets/22033714/69862cc7-dc82-41c6-b394-884815137d74)